railties/lib/rails/application/finisher.rb

```diff
@@ -138,6 +138,9 @@ def self.complete(_state)
           # some sort of reloaders dependency support, to be added.
           require_unload_lock!
           reloader.execute
+          if app.config.eager_load
+            app.routes.eager_load!
```

Comment:
I'd think about moving the `eager_load` config into the routes reloader, and having `execute` do the eager load, instead: it's not a different thing a caller might ask it to do, it's a different set of behaviour for it to perform, based on configuration, when told to reload.